### PR TITLE
RR-646 - Previous Work Experience flow from Previous Work Experience …

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR introduces the concept of a "page flow" to allow page progression from Previous Work Experience Types to Previous Work Experience Details

Best demonstrated with a screen recording ....

https://github.com/ministryofjustice/hmpps-education-and-work-plan-ui/assets/94835226/e041b458-8974-421e-80bb-c855fe9c8acd

### CODE REVIEW

File: server/@types/express/index.d.ts [hunk 1]
```diff
@@ -34,7 +35,6 @@ declare module 'express-session' {
     workedBeforeForm: WorkedBeforeForm
     previousWorkExperienceTypesForm: PreviousWorkExperienceTypesForm
     previousWorkExperienceDetailForm: PreviousWorkExperienceDetailForm
-    previousWorkExperienceForm: PreviousWorkExperienceForm
```
Comment (nathanrussell-moj-digital) [inline server/@types/express/index.d.ts:37]: A hang over from when Previous Work Experience was just one form

File: server/routes/induction/update/previousWorkExperienceDetailUpdateController.test.ts [hunk 2]
```diff
@@ -194,221 +197,351 @@ describe('previousWorkExperienceDetailUpdateController', () => {
...
+          expectedUpdatedPreviousWorkExperienceDetail,
+        )
+
+        expect(inductionService.updateInduction).toHaveBeenCalledWith(prisonNumber, updateInductionDto, 'some-token')
+        expect(res.redirect).toHaveBeenCalledWith(`/plan/${prisonNumber}/view/work-and-interests`)
+        expect(req.session.previousWorkExperienceDetailForm).toBeUndefined()
+        expect(req.session.inductionDto).toBeUndefined()
+      })
+
+      it('should update induction in session but not call API given a PageFlowQueue that is not on the last page', async () => {
```
Comment (mattwills-moj-digital) [inline server/routes/induction/update/previousWorkExperienceDetailUpdateController.test.ts:501]: Good tests 🙌

File: server/routes/induction/update/previousWorkExperienceTypesUpdateController.ts [hunk 3]
```diff
@@ -82,29 +84,45 @@ export default class PreviousWorkExperienceTypesUpdateController extends Previou
...
-    )
+  buildPageFlowQueue = (
+    previousWorkExperienceTypes: Array<TypeOfWorkExperienceValue>,
+    prisonNumber: string,
+  ): PageFlowQueue => {
+    const previousWorkExperienceTypesPageUrl = `/prisoners/${prisonNumber}/induction/previous-work-experience`
+    const pageUrls = [
+      previousWorkExperienceTypesPageUrl,
+      ...previousWorkExperienceTypes.map(
+        workType => `/prisoners/${prisonNumber}/induction/previous-work-experience/${workType.toLowerCase()}`,
```
Comment (mattwills-moj-digital) [inline server/routes/induction/update/previousWorkExperienceTypesUpdateController.ts:119]: Nice 🙌

**Review by mattwills-moj-digital — APPROVED** [review]
This looks really good. A lot of inherent complexity here, but your code is clear and readable, and explained well with comments. It's a yes from me ✅

**Pull request merged**